BD-1271: Button Actions (#3318)
braze-inc/braze-docs: _docs/_developer_guide/platform_wide/braze_actions.md

---
nav_title: Braze Actions Deeplinks
article_title: Braze Actions Deeplink
page_order: 100
description: "Use the `brazeActions://` deeplinks to perform SDK actions within messaging channel buttons"
hidden: true
---

# Braze Actions

> Braze Actions allow you to set up In-App Message or Content Card buttons to perform native SDK functionality.

Author: ```suggestion
Braze Actions let you use "deeplinks" to perform native SDK functionality.

The Braze dashboard includes a few out-of-the-box actions (Request Push Permission, Log Custom Event, and Log Custom Attribute) which can be used in In-App Messages and Content Cards.

For all other actions, or to combine multiple actions, use this guide to construct your own Braze Action deeplink.
```

> For example, you may want a button to request push permission, log a custom event or attribute, or add/remove a user from a specific subscription group.

Author: ```suggestion
```

## Schema

Multiple action `steps` can be included within a `container` action type. A single step without a `container` is also valid.

```json
{
    "type": "container",
    "steps": []
}
```

An individual `step` contains an an action `type`, and optional `args` array:

```json
{
    "type": "logCustomEvent",
    "args": ["event name", {"event": ["properties"]}]
}
```

## URI 

The Braze Actions URI scheme is `brazeActions://v1/{base64encodedJsonString}`

The following javascript shows how to encode/decode the JSON string:

```javascript

>  function decode(encoded) {

Author: ```suggestion
function decode(encoded) {
```

    const binary = atob(encoded)
    const bytes = new Uint8Array(binary.length);
    for (let i = 0; i < bytes.length; i++) {
        bytes[i] = binary.charCodeAt(i);
    }
    return String.fromCharCode(...new Uint16Array(bytes.buffer));
}

function encode(string) {
    const codeUnits = new Uint16Array(string.length);
    for (let i = 0; i < codeUnits.length; i++) {
        codeUnits[i] = string.charCodeAt(i);
    }
    return btoa(String.fromCharCode(...new Uint8Array(codeUnits.buffer))).replace(/=/g, '');
}
```

## Supported Actions

|Type|Args|
|--|--|
|`container`|1. An array of other actions to perform|

> |`logCustomEvent`|1. `event name`<br>2. `event properties` (optional)|

Author: ```suggestion
|`logCustomEvent`|1. `event name`<br>2. `event properties JSON object` (optional)|
```

---
nav_title: Braze Actions Deeplinks
article_title: Braze Actions Deeplink
page_order: 100
description: "Use the `brazeActions://` deeplinks to perform SDK actions within messaging channel buttons"

> hidden: true

Reviewer: @davidbielik is this meant to be hidden? If so I can give it an alias link.

Author: Hi @bre-fitzgerald yes it's meant to be hidden. What's an alias link?

Reviewer: It provides a second way to access the link, so if we add `alias: /braze_actions/` to the YAML, users can access this page from `https://www.braze.com/docs/braze_actions/` as well as `https://www.braze.com/docs/developer_guide/platform_wide/braze_actions/` (where it's currently located). Similar to a permalink for docs that live in a weird place, but if we unhide it later it can still be accessed from the alias link and the normal link.

Author: oh cool! i didn't know about that. I don't know it's necessary here, i'm fine with it being in the full path.
